localstack/utils/aws/message_forwarding.py

```diff
@@ -291,8 +292,17 @@ def add_api_destination_authorization(destination, headers, event):
     return endpoint
 
 
-def add_http_parameters(http_parameters: Dict, endpoint: str, headers: Dict, body):
-    headers.update(http_parameters.get("HeaderParameters", {}))
+def add_target_http_parameters(http_parameters: Dict, endpoint: str, headers: Dict, body):
     endpoint = add_path_parameters_to_url(endpoint, http_parameters.get("PathParameterValues", []))
-    endpoint = add_query_params_to_url(endpoint, http_parameters.get("QueryStringParameters", {}))
+
+    query_params = http_parameters.get("QueryStringParameters", {})
+    prev_query_params = extract_query_string_params(endpoint)[1]
+    query_params.update(prev_query_params)
```

Comment:
I think we should add a small note/reference here about the reason of this change/code block. It's not very clear just by looking at the code what (or more importantly why) it's prioritizing these.